core/unit_test/TestViewAPI.hpp

```diff
@@ -1501,9 +1495,12 @@ class TestViewAPI {
       // quickly.
       if (msg.find("is not a valid size") != std::string::npos) {
         ASSERT_PRED_FORMAT2(::testing::IsSubstring, "is not a valid size", msg);
-      } else {
-        // Otherwise, there has to be some sort of "insufficient memory" error
+      } else if (msg.find("insufficient memory") != std::string::npos) {
         ASSERT_PRED_FORMAT2(::testing::IsSubstring, "insufficient memory", msg);
+      } else {
+        // Otherwise, there has to be some sort of "unknown error" error
+        ASSERT_PRED_FORMAT2(::testing::IsSubstring,
+                            "because of an unknown error.", msg);
```

Comment:
For `SYCL` can't provide more details why the memory allocation failed so "unknown error" is the only option here. That being said, I think it's fine to also allow this error case here. It clearly also comes from within the library.

Reply:
Not sure how I feel about that change.  I would use an `#ifdef KOKKOS_ENABLE_SYCL` here.